sympy/solvers/solveset.py

```diff
@@ -366,26 +366,33 @@ def _solve_as_rational(f, symbol, domain):
 def _solve_trig(f, symbol, domain):
     """ Helper to solve trigonometric equations """
     f = trigsimp(f)
-    f_original = f
+    unsolved = ConditionSet(symbol, Eq(f, 0), domain)
     f = f.rewrite(exp)
     f = together(f)
-    g, h = fraction(f)
+    G, H = [i.expand() for i in fraction(f)]
     y = Dummy('y')
-    g, h = g.expand(), h.expand()
-    g, h = g.subs(exp(I*symbol), y), h.subs(exp(I*symbol), y)
-    if g.has(symbol) or h.has(symbol):
-        return ConditionSet(symbol, Eq(f, 0), S.Reals)
-
-    solns = solveset_complex(g, y) - solveset_complex(h, y)
-
-    if isinstance(solns, FiniteSet):
-        result = Union(*[invert_complex(exp(I*symbol), s, symbol)[1]
-                       for s in solns])
-        return Intersection(result, domain)
-    elif solns is S.EmptySet:
-        return S.EmptySet
-    else:
-        return ConditionSet(symbol, Eq(f_original, 0), S.Reals)
+    result = unsolved
+    for do in range(2):
+        if do == 0:
+            cov = exp(I*symbol)
+        elif do == 1:
+            cov = exp(symbol)
+        g, h = [i.subs(cov, y) for i in (G, H)]
+        if g.has(symbol) or h.has(symbol):
+            continue
+        # the change of variables succeeded
+        soln = solveset_complex(g, y) - solveset_complex(h, y)
+        if isinstance(soln, FiniteSet):
+            result = Union(*[
+                invert_complex(cov, s, symbol)[1] for s in soln])
+            result = Intersection(result, domain)
```

Comment:
is it necessary to use the domain here or will that be taken care of by solveset?